### DISCUSSION TITLE
SCD30 Driver

### DISCUSSION THREAD

**Discussion by MarkusGaelli** [toitlang/toit#185] [Drivers]
Hi 

Wanted to create a driver for the SCD30 environmental sensor, but @qvisten999 beat me here. :-)
Managed to get it running but puzzled about a few things:
Should I import the lib like this?
![2021-12-22_16-14-55](https://user-images.githubusercontent.com/1045437/147118417-e4228378-e91d-49e6-b316-d9d9f7ed7c7d.jpg)
How would I correct the "deprecation"?
Why do I need to extend it with "driver" and how could I "show" the class Scd30 directly?

Cheers
Markus

**Comment by lask** [comment]
Hi Markus

Good work on the driver. I hope you will publish it to the package registry, so others can use it. 

AFAIR, when you write
```
import scd30_driver.driver
```
then you need to use the suffix when you refer to things from the package, i.e. `driver`. If you prefer to use something else you can use `as`. For instance
```
import scd30_driver.driver as scd30
```

**Reply by lask** [reply]
I was on my phone and only read half of the question. Give me a second and I will answer again.

**Reply by MarkusGaelli** [reply]
Thx, but to clarify: The driver was written by @qvisten999 -- see https://github.com/qvisten999/scd30
I wrote one here but not for toit but for [microblocks](https://microblocks.fun/): https://bitbucket.org/markus-spielt/smallvm/src/dev/gp/Libraries/Sensing/SCD30.ubl

I see you were on your phone, so no worries, just wonder how to import things correctly...

**Reply by lask** [reply]
Ah, I see. @floitsch answer is very comprehensive, so I will just add that you can also just show the `Scd30` class with the `show` keyword:
```
import scd30_driver.driver show Scd30
```

**Comment by floitsch** [comment]
This warning is because an import of the form `import foo.bar.gee` gives a prefix `gee`.
In the old days it used to be `foo`, and for backwards compatibility it still does, but provides the warning that some code later on uses it this way and that it should be fixed.

In your example there is a line `scd30_driver.Scd30` which is the reason for the use of the prefix.

As Lau correctly states, you could either use the prefix `driver`, or change the prefix to `scd30`. Alternatively, you could also just use `show *` to get rid of the prefix altogether.

For what it's worth, you shouldn't need to worry about this as much as normally an `import scd30` would do the trick.
However, the `scd30` package has a small unfortunate mistake. It declares that it's package name is scd30_driver, instead of `scd30`. I will try to fix this, but for now you could work around this by installing the package this way:
```
toit pkg install --name=scd30 https://github.com/qvisten999/scd30
# or, for the OSS version
toitpkg pkg install --name=scd30 https://github.com/qvisten999/scd30
```

This way the name of the driver would be scd30, and an `import scd30` would be equivalent to `import scd30.scd30`.

**Reply by MarkusGaelli** [reply]
Thanks Florian!

**Reply by floitsch** [reply]
Here is my PR to fix the name.
https://github.com/qvisten999/scd30/pull/1

**Reply by floitsch** [reply]
Explanation for the name: if the package.yaml doesn't specify a name, then the title of the README is used.
See the package quickstart: https://docs.toit.io/language/package/pkgguide/#creating-a-package

**Comment by jacobqvist** [comment]
Hi @MarkusGaelli 
Sorry for the late reply. Updated the driver with the changes from @floitsch. https://pkg.toit.io/package/github.com%2Fqvisten999%2Fscd30@v1.0.1

Merry Christmas
